### ISSUE TITLE
stdweb always included as a dependency no matter the platform

### ISSUE THREAD

**Issue by arturoc** [tomaka/cpal#243]
This is probably a bug in cargo but stdweb always gets included as a dependency of this crate. It's usually ok unless you try to compile a dynamic library including this crate in which case the resulting binary will have dependencies with emscripten and fail to load.

Adding an stdweb feature to optionally use stdweb when compiling for the web should solve the problem until cargo does the right thing

**Comment by tomaka** [comment]
Could you check your `Cargo.lock` to see if it's not accidentally one of the other dependencies that pulls `stdweb`? A Cargo bug is pretty unlikely I think.

**Comment by arturoc** [comment]
it's coming from cpal but let me double check with a simpler example just in case

**Comment by arturoc** [comment]
with a new project and no other dependencies but cpal and building in linux 64 (using nightly in case it makes any difference):

Cargo.toml:
```toml
[package]
name = "test_cpal"
version = "0.1.0"
authors = ["arturo castro <arturo@openframeworks.cc>"]

[dependencies]
cpal="*"
```


Cargo.lock:
```
[[package]]
name = "cpal"
version = "0.8.2"
source = "registry+https://github.com/rust-lang/crates.io-index"
dependencies = [
 "alsa-sys 0.1.2 (registry+https://github.com/rust-lang/crates.io-index)",
 "core-foundation-sys 0.5.1 (registry+https://github.com/rust-lang/crates.io-index)",
 "coreaudio-rs 0.9.1 (registry+https://github.com/rust-lang/crates.io-index)",
 "lazy_static 1.1.0 (registry+https://github.com/rust-lang/crates.io-index)",
 "libc 0.2.43 (registry+https://github.com/rust-lang/crates.io-index)",
 "stdweb 0.1.3 (registry+https://github.com/rust-lang/crates.io-index)",
 "winapi 0.3.5 (registry+https://github.com/rust-lang/crates.io-index)",
]
```

**Comment by tomaka** [comment]
I think the dependency is also present in the `Cargo.lock`, but it's built only if necessary.

**Comment by arturoc** [comment]
yeah i imagine since there's also stuff like winapi and coreaudio in linux but i was getting undefined symbols for emscripten functions so it was definitely being included. in any case this only happens when compiling a dynamic library so it's a pretty obscure case

**Comment by roderickvd** [comment]
stdweb no longer is a dependency, at all.
